src/pretix/control/templates/pretixcontrol/event/tickets.html

```diff
@@ -28,9 +28,14 @@ <h1>{% trans "Ticket download" %}</h1>
                 {% for provider in providers %}
                     <div class="panel panel-default ticketoutput-panel">
                         <div class="panel-heading">
-                            <a href="{% url "control:event.settings.tickets.preview" event=request.event.slug organizer=request.organizer.slug output=provider.identifier %}"
-                                    class="btn btn-default btn-sm pull-right {% if not provider.preview_allowed %}disabled{% endif %}"
-                                    target="_blank">
+                            {% if provider.download_action %}
+                                <a href="{{ provider.download_action }}" class="btn btn-default btn-sm pull-right {% if not provider.preview_allowed %}disabled{% endif %}" target="_blank"
```

Comment:
With the changes proposed above, this would not longer work as easily. However, I'm fine with not having previews for non-downloadable providers.